### PR TITLE
Unify handling of the current user of a request

### PR DESCRIPTION
**Pull request by ThiefMaster**
My approach is to first use the slightly ugly solution where `session.user` returns the current user, and then later (in a separate PR) replace `session.user` with something else.

### CODE REVIEW

File: indico/web/util.py [hunk 1]
```diff
@@ -181,31 +185,84 @@ def url_for_index(_external=False, _anchor=None):
...
+
+    url = url_unparse((
+        '',
+        '',
+        parsed.path,
+        url_encode(sorted(params.items()), sort=True),
+        parsed.fragment
+    ))
+    signer = Signer(user.signing_secret, salt='url-signing')
+    return signer.verify_signature(url.encode(), signature)
```
Comment (plourenco) [inline indico/web/util.py:211]: Don't we need to encode the signature as before? (also applies to other occasions below)
```suggestion
    return signer.verify_signature(url.encode(), signature.encode())
```

**Review by plourenco — APPROVED** [review]
Nice, I like the way this is heading 👍

For future reference, we talked about the possibility of keeping the simple assignment `session.user =` instead of the setter, but that behaviour is only meant to be restored later when `session.user` is replaced and setting and reading to the exact same thing.

File: indico/modules/events/registration/util.py [hunk 2]
```diff
@@ -413,7 +413,7 @@ def _build_base_registration_info(registration):
         'registrant_id': str(registration.id),
         'checked_in': registration.checked_in,
         'checkin_secret': registration.ticket_uuid,
-        'full_name': '{} {}'.format(personal_data.get('title', ''), registration.full_name),
+        'full_name': '{} {}'.format(personal_data.get('title', ''), registration.full_name).strip(),
```
[the remark this answers is not in the document]
Comment (ThiefMaster) [reply indico/modules/events/registration/util.py:416]: I spotted it while testing and was too lazy to push it directly to master ;)

File: indico/web/util.py (hunk 1 above)
Comment (ThiefMaster) [reply indico/web/util.py:211]: No, it uses `wants_bytes()` internally in most places. The next itsdangerous version will do that for the data arg as well, but it's still a prerelease...

**Pull request merged**